CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/parser/PCAPImportFileParser.java

```diff
@@ -0,0 +1,836 @@
+        }
+        return InetAddresses.toAddrString(
+                InetAddresses.forString(builder.toString().substring(0, builder.toString().length() - 1)));
+    }
+
+    /**
+     * Open and return the supplied PCAP file, with exception handling wrapper
+     * to ensure exceptions are caught and propagated in a controlled manner.
+     * @param file PCAP file to open
+     * @return opened PCAP file
+     * @throws IOException 
+     */
+    private Pcap openPcap(File file) throws IOException {
+        try {
+                return Pcap.openStream(file);
+            } catch (final Exception ex) {
+                throw new IOException(WARN_INVALID_PCAP);
+            }
+    }
+    
+    /**
+     * Designed to perform common decoding as required by the parse and preview
+     * functions. Perform common extraction of data from PCAP file. This uses
+     * the io.pkts library (where possible) to decode content of PCAP file,
+     * looping over contained packets and extracting key information.
+     *
+     * @param input Input file
+     * @param limit Row limit
+     * @return a List of String arrays, each of which represents a row in the
+     * resulting table.
+     * @throws IOException
+     */
+    private List<String[]> getResults(final InputSource input, final int limit) throws IOException {
+        // Define names of 'columns' that will be extracted from PCAP
+        final String[] headings = {
+            "Frame", "Time",
+            "Src MAC", "Src IP", "Src Port", "Src Type",
+            "Dest MAC", "Dest IP", "Dest Port", "Dest Type",
+            "Ethertype", "Protocol", "Length", "Info"};
+
+        // Define results object to be used and add headings, prior to looping
+        // through and adding row data
+        final List<String[]> results = new ArrayList<>();
+        results.add(headings);
+
+        try {
+            Pcap pcap = openPcap(input.getFile());
+            frameCounter = 1;
+
+            // Use io.pkts fucntionality to extract rawe packets and iterate
+            // over them. Breaking out individual payload fields is not using
+            // io.pkts as its quite restrictive, this is instead being done on
+            // a bespoke basis. At some point, there may be a case to try and
+            // remove all dependencies on io.pkts and manually loop through
+            // PCAP data frames
+            pcap.loop(new PacketHandler() {
+                @Override
+                public boolean nextPacket(final Packet packet) throws IOException {
+
+                    // Extract timestamp and frame counter strings
+                    final SimpleDateFormat formatter =
+                            new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS z");
+                    // Set a default timezone to ensure consistant times across multiple locations
+                    // Currently set to GMT +0
+                    formatter.setTimeZone(TimeZone.getTimeZone("GMT"));
```

Comment:
not 100% sure but you could replace `formatter` with `TemporalFormatter.UTC_DATE_TIME_WITH_MILLISECONDS_FORMATTER` if that is the format you are after